bin/old-search-function-test-utilizing-llm.py

```diff
@@ -0,0 +1,93 @@
+import os
+import sys
+import json
+from datetime import date
+
+import ailab.db as db
+from ailab.models import openai
+import ailab.db.finesse as finesse
+
+from ailab.db.finesse.test_queries import get_random_chunk
+
+TEST_VERSION = date.today()
+WANTED_GENERATED_QUESTIONS = 5
+CHARACTER_LIMIT = 14383
+
+
+def main():
+    if len(sys.argv) < 2:
+        print("Usage: " + sys.argv[0] + " PROMPT_PATH")
+        print("PROMPT_PATH: Directory containing the API prompt")
+        sys.exit(1)
+    prompt_path = sys.argv[1]
+    if not os.path.exists(prompt_path):
+        print(f"The directory '{prompt_path}' does not exist.")
+        sys.exit(1)
+
+    system_prompt = finesse.load_prompt(prompt_path, "qna_system_prompt.txt")
+    user_prompt = finesse.load_prompt(prompt_path, "qna_user_prompt.txt")
+    json_template = finesse.load_json_template(prompt_path)
+
+    project_db = db.connect_db()
+
+
+    AVERAGE_TOKENS_BY_CHUNK = 0
+
+    
+    for i in range(WANTED_GENERATED_QUESTIONS):
+        random_chunk = ""
+
+        # Fetch a random chunk from the database
+        with project_db.cursor() as cursor:
+            random_chunk = get_random_chunk(cursor)
```

Comment:
This is the old version of the code I kept because I was still working on it. It has been deleted now, and this adjustment has been made in the new version of the script. More precisely here: [9e4aa52](https://github.com/ai-cfia/ailab-db/pull/47/commits/9e4aa52df7007e4734f04ea6beb341f5584ac880)

Not sur to fully understand what was needed, but this is an upgrade, and now the cursor only open once.